test/run_clippy.sh

```diff
@@ -118,10 +119,10 @@ add_lints_to_clippy_args() {
   done
 }
 
-add_lints_to_clippy_args -A "${unexamined_lints[@]}"
-add_lints_to_clippy_args -A "${allowed_lints[@]}"
-add_lints_to_clippy_args -W "${lints_to_fix[@]}"
-add_lints_to_clippy_args -D "${denied_lints[@]}"
+add_lints_to_clippy_args -A "${unexamined_lints[@]+"${unexamined_lints[@]}"}"
+add_lints_to_clippy_args -A "${allowed_lints[@]+"${allowed_lints[@]}"}"
+add_lints_to_clippy_args -W "${lints_to_fix[@]+"${lints_to_fix[@]}"}"
+add_lints_to_clippy_args -D "${denied_lints[@]+"${denied_lints[@]}"}"
```

Comment:
What do these changes do?

Reply:
Allows for empty arrays. I'm totally open to other possibilities, but this allows us to essentially pass an empty string in the event that there is a 0-element array, which throws an unused error.

Reply:
That shouldn't be necessary. `add_lints_to_clippy_args` modifies an array `clippy_args`, not a string.

Reply:
This is the error I was looking to avoid: https://buildkite.com/chef/habitat-sh-habitat-master-verify/builds/581#a7a0d244-034a-4097-84f2-008562c13102

```
info: downloading component 'clippy'
  | info: installing component 'clippy'
  | test/run_clippy.sh: line 122: unexamined_lints[@]: unbound variable
  | make: *** [lint] Error 1
```

Reply:
That's weird. [Is this the exact version of `run_clippy.sh`](https://github.com/habitat-sh/habitat/blob/513f23e0fa58590a8129d183a726d70e4bbd9eb3/test/run_clippy.sh) that was used in that run? In that version, `unexamined_lints` isn't unbound, [it's initialized to an empty array on line 39](https://github.com/habitat-sh/habitat/blob/513f23e0fa58590a8129d183a726d70e4bbd9eb3/test/run_clippy.sh#L39).

I try this locally and it's not a problem:
```bash
➤ bash -l
$ set -eou pipefail
$ unexamined_lints=()
$ clippy_args=()
$
$ add_lints_to_clippy_args() {
>   flag=$1
>   shift
>   for lint
>   do
>     clippy_args+=("$flag" "${lint}")
>   done
> }
$ echo "${unexamined_lints[@]}"

$ add_lints_to_clippy_args -A "${unexamined_lints[@]}"
$ echo $?
0
```